### PR TITLE
Make Toffoli an atomic, leaf bloq

### LINKED ISSUE #1386: Make `Toffoli` a leaf bloq
Specific changes that are needed to do this part of https://github.com/quantumlib/Qualtran/issues/873

### PR DESCRIPTION
**Pull request by mpharrigan**
Fixes #1386 , which is part of the roadmap item #873 

Toffoli's should be counted directly. There's no existing circuit for doing the gate with 4 toffolis, but one can be added later as a bespoke method. 

All of the test changes are very straightforward: many of them were trying to test toffoli count anyways :)

### CODE REVIEW

File: qualtran/bloqs/chemistry/sf/prepare_test.py [hunk 1]
```diff
@@ -66,13 +63,14 @@ def test_outerprep_t_counts():
...
-    cost1a_mod = prep.call_graph()[1][TGate()] // 4
-    cost1a_mod += prep.adjoint().call_graph()[1][TGate()] // 4
+    cost1a_mod = get_cost_value(prep, QECGatesCost()).total_t_count(ts_per_cswap=4) // 4
+    cost1a_mod += get_cost_value(prep.adjoint(), QECGatesCost()).total_t_count(ts_per_cswap=4) // 4
     assert cost1a != cost1a_mod
     bL = nb_l + num_bits_state_prep + 2
     cost1b = QR(num_aux + 1, bL)[-1] + QI(num_aux + 1)[-1]
     cost1cd = 2 * (num_bits_state_prep + nb_l + 1)
     of_cost = cost1a_mod + cost1b + cost1cd
+    toff -= 1  # TODO: unknown difference porting to QECGatesCost().
```
Comment (mpharrigan) [inline qualtran/bloqs/chemistry/sf/prepare_test.py:73]: @fdmalone can you help me investigate this discrepancy?

**Comment by mpharrigan** [comment]
Offline @fdmalone told me the discrepancy shouldn't block this. Opening an issue to track

File: qualtran/bloqs/chemistry/sf/prepare_test.py (hunk 1 above)
Comment (mpharrigan) [reply qualtran/bloqs/chemistry/sf/prepare_test.py]: #1391

**Review by fdmalone — APPROVED** [review]
LGTM

**Pull request merged**